src/realm/dictionary.cpp

```diff
@@ -394,7 +399,53 @@ void Dictionary::sort(std::vector<size_t>& indices, bool ascending) const
         });
     }
 }
-void Dictionary::distinct(std::vector<size_t>&, util::Optional<bool>) const {}
+
+void Dictionary::distinct(std::vector<size_t>& indices, util::Optional<bool> ascending) const
+{
+    align_indices(indices);
+    if (ascending) {
```

Comment:
Can't this be done the same way as in `Lst<T>::distinct`? Then you don't have to use `std::unordered_set`